### PR TITLE
test: add go unit tests for experiment bulk actions [ET-138]

### PR DESCRIPTION
**Pull request by jesse-amano-hpe**
## Ticket

[ET-138]
[ET-139]
[ET-140]

## Description

Adds unit tests to cover `ActivateExperiments`, `CancelExperiments`, `KillExperiments`, `PauseExperiments`, and the underlying `getExperimentsEditableByUser` logic which is mostly orthogonal to the "do this operation on every eligible experiment" piece.

## Test Plan

No new behaviors have been added; this is an extremely light refactor for testability and the rest of it is for test coverage only.

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[ET-138]: https://hpe-aiatscale.atlassian.net/browse/ET-138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-139]: https://hpe-aiatscale.atlassian.net/browse/ET-139?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-140]: https://hpe-aiatscale.atlassian.net/browse/ET-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/experiment/bulk_action_test.go [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+		expectedErr           bool
+	}{
+		{
+			name:                "user login error",
+			args:                args{},
+			experimentsEditable: experimentsEditable{err: errors.New("user has no permision to edit experiments")},
+			expectedErr:         true,
+		},
+		{
+			name: "three experiments selected, one found",
```
Comment (AmanuelAaron) [inline master/internal/experiment/bulk_action_test.go:89]: nit: technically 2 are found but one of the two is in an invalid state

File: master/internal/experiment/bulk_action_test.go [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+		filters       *apiv1.BulkExperimentFilters
+	}
+	type experimentsEditable struct {
+		expIDs []int32
+		err    error
+	}
+
+	ctx := context.Background()
+
+	tests := []struct {
```
Comment (AmanuelAaron) [inline master/internal/experiment/bulk_action_test.go:426]: is there a case for when the experiment is already Paused?

File: master/internal/experiment/bulk_action_test.go [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+					Error: status.Errorf(codes.FailedPrecondition, "experiment in terminal state"),
+					ID:    132,
+				},
+				{
+					ID: 142,
+				},
+			},
+		},
+		{
+			name: "filters are used",
```
Comment (AmanuelAaron) [inline master/internal/experiment/bulk_action_test.go:114]: Perhaps more test cases on different filter fields?

**Review by AmanuelAaron — COMMENTED** [review]
Tests look great! Left a few comments, but overall really good.

**Review by hamidzr — APPROVED** [review]

File: master/internal/experiment/bulk_action_test.go (hunk 3 above)
Comment (jesse-amano-hpe) [reply master/internal/experiment/bulk_action_test.go:114]: This has proved trickier than I hoped, and I keep getting pulled off for other issues while trying to make these test cases, which then makes all these tickets bleed into more sprints. I think I'd like to just merge this, and make a new ticket for the additional coverage, if that's OK with you.

**Pull request merged**